File: untitled/src/main/katas/lupe/codewars/katas/java/squareSum/SquareSum.java
package lupe.codewars.katas.java.squareSum;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SquareSum {
    public int squareSum(int[] numbers) {
        if (numbers.length == 0) {
            return 0;
        }
        int initialValue = 0;
        List<Integer> numbersList = new ArrayList<>();
        Arrays.stream(numbers).forEach(numbersList::add);
        return numbersList.stream()
                .map(el -> el * el)
                .reduce(initialValue, Integer::sum);
    }
}

File: java-kotlin-katas/src/main/katas/lupe/codewars/katas/kotlin/squareSum/SquareSum.kt
package lupe.codewars.katas.kotlin.squareSum

class SquareSum {

    fun squareSum(numbers: IntArray): Int {
        return if (numbers.isEmpty()) 0
            else numbers
                .map { el -> el * el }
                .reduce { acc: Int, current: Int -> acc + current }
    }
}
